AutoRest/Generators/NodeJS/NodeJS.Tests/typings/mocha/mocha.d.ts

```diff
@@ -0,0 +1,214 @@
+// Type definitions for mocha 2.2.5
+// Project: http://mochajs.org/
```

Comment:
Yes, typings should be updated if changing the mocha version.   This should work to do that:
`tsd install mocha --save --overwrite`
For taking dependencies on a new package where want type checking this should do it:
`tsd install <package> --save`
See https://github.com/DefinitelyTyped/tsd for full doc.
Basically `tsd.json` stores the type dependencies (much like `package.json` for node dependencies) and `typings` is generated from that containing all the type files (like `node_modules`).